app/src/components/UserForm.tsx

```diff
@@ -0,0 +1,182 @@
+"use client";
+
+interface UserFormProps {
+  formType: "signup" | "signin";
+}
+
+import { useState } from "react";
+import "../app/globals.css";
+import Link from "next/link";
+import { Button, buttonVariants } from "./ui/button";
+import { cn } from "../lib/utils";
+
+interface UserFormProps {
+  formType: "signup" | "signin";
+}
+
+const UserForm = (props: UserFormProps) => {
+  const { formType } = props;
+
+  const isSignup = formType === "signup";
+
+  const [formData, setFormData] = useState({
```

Comment:
please use `react-hook-form` and `zod` packages to handle form related tasks 
refer to [lingo.ai](https://github.com/joshsoftware/lingo.ai/blob/main/app/src/components/UserForm.tsx)

Reply:
done